pkg/config/config.go

```diff
@@ -84,8 +84,15 @@ func getOcSession(pod, container, namespace string, timeout time.Duration, optio
 		gomega.Expect(err).To(gomega.BeNil())
 		// Set up a go routine which reads from the error channel
 		go func() {
-			err := <-outCh
-			gomega.Expect(err).To(gomega.BeNil())
+			for {
+				select {
+				case err := <-outCh:
+					log.Errorf("SpawnOc error: %v from %s %s", err, oc.GetPodName(), oc.GetPodContainerName())
+					gomega.Expect(err).To(gomega.BeNil())
```

Comment:
Needs a previous defer ginkgo.GinkgoRecover() to avoid panic'ing in case err is not nil.

Reply:
Hmm, I was wondering whether the original assertion is really the appropriate action here. Recover would let the test run continue, right? I don't think that's right either. Is there an alternative to panic if we just want to stop the test run?

Reply:
gingko.Fail() will only fail the test and continue, unless you have fast fail, in which case, the test would stop on the first failure.

Reply:
Panic is the only way then? Personally I don't mind. It just looks a bit scary to people who don't know what it means. I could improve the log text a bit so they understand that the panic is intentional?

Reply:
I'm not sure what are you trying to achieve here, but here's what to expect :
 * a failure in gomega or ginkgo.Fail would result in current test (ginkgo.It ) be stopped and the code will move to next test.
 * panic will stop everything.

Reply:
I think we should stop the test altogether, as the test result wouldn't be useful if we continue running with broken sessions.

Reply:
Problem is this code is not part of any test: it's run inside the autodiscovery mechanism in the BeforeEach section. So, if the oc shell cannot be created, even using the defer ginkgo.GinkgoRecover() will make the first TC in the test suite and those after any intrusive TC to fail.
I'd go with removing the gomega assertion and use a logrus.Warn/Error message. These ocSessions are only used for the connectivity TCs, right?

Reply:
No, they are used by other tests too.